tools/distribution/verify.py

```diff
@@ -0,0 +1,203 @@
+import argparse
+import pathlib
+import sys
+from functools import cached_property
+from typing import Optional, Type
+
+import aiodocker
+
+from tools.base import checker, utils
+from tools.distribution import distrotest
+
+ENVOY_MAINTAINER = "Envoy maintainers <envoy-maintainers@googlegroups.com>"
+ENVOY_VERSION = "1.20.0"
+
+
+class PackagesConfigurationError(Exception):
+    pass
+
+
+class PackagesDistroChecker(checker.AsyncChecker):
+    _active_distrotest = None
+    checks = ("distros",)
+
+    @property
+    def active_distrotest(self) -> Optional[distrotest.DistroTest]:
+        """Currently active test"""
+        return self._active_distrotest
+
+    @cached_property
+    def config(self) -> dict:
+        """Config parsed from the provided path
+
+        Expects a yaml file with distributions in the following format:
+
+        ```yaml
+
+        debian_buster:
+          image: debian:buster-slim
+          ext: buster.changes
+
+        ubuntu_foo:
+          image: ubuntu:foo
+          ext: foo.changes
```

Comment:
What is this `ext` and `changes`?

Reply:
`ext` is the file extension and in the case of deb files we want the `$distro.changes` file for a particular distro (when installing)

ill add some commentage